pkg/storages/s3/uploader.go

```diff
@@ -74,8 +77,16 @@ func (uploader *Uploader) createUploadInput(bucket, path string, content io.Read
 }
 
 func (uploader *Uploader) upload(bucket, path string, content io.Reader) error {
+	var err error
+	uploadTimeout := uploader.uploadTimeout
 	input := uploader.createUploadInput(bucket, path, content)
-	_, err := uploader.uploaderAPI.Upload(input)
+	if uploadTimeout != 0 {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(uploadTimeout * int(time.Second)))
```

Comment:
This part repeats too many times, just move it to some function, e.g. `configureUploadContext`